### PR TITLE
ci: add smoke test workflow

### PR DESCRIPTION
**Pull request by G-Rath**
Came across this when picking up work for `eslint-plugin-jest-dom` - seems like a pretty cool thing to have for us too.

### CODE REVIEW

File: .github/workflows/smoke-test.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - run: |
+          npm install
```
Comment (SimenB) [inline .github/workflows/smoke-test.yml:17]: Npm?

File: .github/workflows/smoke-test.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+          npm run build
+          npm link
+          npm link eslint-plugin-jest
+      - uses: AriPerkkio/eslint-remote-tester-run-action@v2
+        with:
+          issue-title: 'Results of weekly scheduled smoke test'
+          eslint-remote-tester-config: smoke-test/eslint-remote-tester.config.js
```
Comment (SimenB) [inline .github/workflows/smoke-test.yml:24]: This file is missing?

**Comment by G-Rath** [comment]
@SimenB yeah I merged this by mistake 😅 have fixed in #1003

File: .github/workflows/smoke-test.yml (hunk 1 above)
Comment (G-Rath) [reply .github/workflows/smoke-test.yml:17]: Yeah I can't get `yarn link` working - happy if you want to have a crack at it

Comment (AriPerkkio) [reply .github/workflows/smoke-test.yml:17]: I think yarn had some issues to get package linking working across host and docker container. Same issues were seen with older npm versions: https://github.com/yannickcr/eslint-plugin-react/pull/3120#issuecomment-959089250.

**Pull request merged**